core/src/main/java/org/opensearch/sql/planner/DefaultImplementor.java

```diff
@@ -129,7 +130,13 @@ public PhysicalPlan visitValues(LogicalValues node, C context) {
 
   @Override
   public PhysicalPlan visitLimit(LogicalLimit node, C context) {
-    return new LimitOperator(visitChild(node, context), node.getLimit(), node.getOffset());
+    PhysicalPlan child = visitChild(node, context);
+    // Optimize sort + limit to take ordered operator
+    if (child instanceof SortOperator sortChild) {
```

Comment:
have u consider add a rule in LogicalPlanOptimizer?

Reply:
Yeah, I've considered doing this optimization via a rule in logical plan in the beginning, then I should also add an new logical operator like LogicalTakeOrdered to replace LogicalLimit + LogicalSort. 

But I find it may be not proper to do that because this is an physical execution level optimization. The logical plan of PPL or SQL is not only able to be executed on OpenSearch,  it could also be translated to execute on other engine like Spark(opensearch-spark supports this) or something else in the future. But usually, they don't have such logical or physical operator like TakeOrdered and will make this logical plan incompatible.

I think this optimization is all about execution and deeply bound to execution engine, so I choose to do that replacement in physical plan.

Reply:
make sense.